### DISCUSSION TITLE
Modbus read_holding_registers()

### DISCUSSION THREAD

**Discussion by johanpza** [StephanJoubert/home_assistant_solarman#338] [Ideas]
Hi

Great repo, thanks. 

Are there any plans for adding a read service, read_holding_registers() for example?
https://pysolarmanv5.readthedocs.io/en/stable/

Thanks

**Comment by StephanJoubert** [comment]
Hi Johan. 

I have only little time to spend on the project, but found the time to integrate with Pysolarman and also added the service call to write_holding_register. 

@jmccrohan  If you would review the change and make updates as needed , it would be appreciated. I plan to make a new release in the next few days after letting it run for a while.

**Reply by jmccrohan** [reply]
@StephanJoubert Thanks! I have just installed the latest `HEAD` and will test for a few days. I will review the code and prepare a PR with any changes.

**Comment by johanpza** [comment]
HiThat is great, thank you very much. Cheers On 04 Aug 2023, at 18:47, StephanJoubert ***@***.***> wrote:﻿
Hi Johan.
I have only little time to spend on the project, but found the time to integrate with Pysolarman and also added the service call to write_holding_register.
@jmccrohan  If you would review the change and make updates as needed , it would be appreciated. I plan to make a new release in the next few days after letting it run for a while.
